core/spec/models/spree/product_spec.rb

```diff
@@ -152,8 +152,8 @@ class Extension < Spree::Base
       context "with currency set to JPY" do
         before do
           product.master.default_price.currency = 'JPY'
-          product.master.default_price.save!
           stub_spree_preferences(currency: 'JPY')
+          product.master.default_price.save!
```

Comment:
Why do we need this change?

Reply:
`default_price` was an association with `autosave` option set to `true`, same that `master` on `product`. Now we need to save it manually.

Reply:
Thinking about it twice, maybe that could have backward compatibility issues. However, at least, not providing the price will result [in a validation error](https://github.com/solidusio/solidus/blob/58f2e0bbcc6c27cff459985bce67e8912f324220/core/app/models/spree/product.rb#L111) when the app is configured to require a default price. I don't think there's a reliable way to detect the situation to emit a warning, though.

Reply:
But the change is only moving the save a line above, isn't it?

Reply:
I'm sorry, I don't understand what you mean. We're adding a new line to the test so that price gets persisted into the database.

Reply:
This is what I see in the diff:

<img width="692" alt="CleanShot 2021-06-01 at 09 36 22@2x" src="https://user-images.githubusercontent.com/167946/120284722-d8b03980-c2bc-11eb-9791-795ee2395dd4.png">

Reply:
Following @kennyadsl suggestion, we updated it a bit to look less confusing:

```ruby
product.master.default_price.update!(currency: 'JPY')
stub_spree_preferences(currency: 'JPY')
```